### PR TITLE
Mirrors now display your reflection. (Remake of #49833)

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Adds a new component that reflects movables on the parent which mirrors now use.
It works good and all, alas it currently doesn't use either connect_loc or proximity monitors to listen to the turfs around but its own procs, because connect_loc is limited to the location of the movable and the proximity monitor is crappy old, doesn't support datum proximity receivers or moving things very well and would probably only end up bloating the component. So uh, maybe if you know an easy solution to this... meanwhile I'll make this a draft PR.

## Why It's Good For The Game
Making mirrors cooler. Here the pic from #49833:
immagine 
and here another from this remake, proof it works: 
Bolshevik Spess Junk LV 2021-11-07 024750
## Changelog

:cl: Ghommie (original PR by bandit)
expansion: Nanotrasen has upgraded mirrors to actually function as mirrors.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Despite everything, it's still you.

**Comment by ATH1909** [comment]
bikeshed suggestion: make vampires not show up in mirror reflections

File: code/datums/components/reflection.dm [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		return
+	if(old_loc)
+		for(var/turf/target_turf as anything in range(1, old_loc))
+			UnregisterSignal(target_turf, list(COMSIG_ATOM_ENTERED, COMSIG_ATOM_EXITED))
+	reflection_effect.vis_contents -= reflected_movables
+	LAZYNULL(reflected_movables)
+	if(isturf(source.loc))
+		set_reflection_targets()
+
+/// The component will call register_reflection_turf() on turfs that pass certain conditions. Called when the
```
Comment (Kylerace) [inline code/datums/components/reflection.dm:90]: when the

File: code/game/objects/structures/mirror.dm [hunk 2]
```diff
@@ -27,8 +28,22 @@
...
+/obj/structure/mirror/Destroy()
+	QDEL_NULL(reflection_comp)
+	return ..()
+
+/obj/structure/mirror/proc/add_reflection_comp()
+	if(reflection_comp)
+		return
+	var/list/mask_filter = alpha_mask_filter(icon = icon('icons/obj/watercloset.dmi', "mirror_mask"))
+	var/matrix/matrix = matrix(-0.75, 0, 0, 0, 0.75, 0)
+	reflection_comp = AddComponent(/datum/component/reflection, use_parent_dir = TRUE, reflection_filter = mask_filter, reflection_matrix = matrix)
```
Comment (Kylerace) [inline code/game/objects/structures/mirror.dm:46]: why are you storing a var to the component? you dont appear to be doing anything to it other than deleting it on Destroy() and that behavior should be done by the component

Comment (Ghommie) [reply code/game/objects/structures/mirror.dm:46]: The component is also removed when the mirror is broken and readded once it's been fixed. See lines 110 and 139.

**Comment by zxaber** [comment]
Technology

**Comment by itseasytosee** [comment]
> bikeshed suggestion: make vampires not show up in mirror reflections

#48796

File: code/game/objects/structures/mirror.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/game/objects/structures/mirror.dm:46]: You should use LoadComponent for the fixing and use signals to react to the breaking.
You can handle the signal stuff in the mirror and just have a delete proc to register onto, but you shouldn't be storing the component like a var.

**Comment by LemonInTheDark** [comment]
:sadkirby:

**Comment by Ghommie** [comment]
> :sadkirby:

Don't worry, we'll reopen this once the proximity monitor refactor is done.

**Comment by LemonInTheDark** [comment]
:sadderkirby:

**Comment by Ghommie** [comment]
... once the feature freeze is over.

**Comment by ChungusGamer666** [comment]
> > :sadkirby:
> 
> Don't worry, we'll reopen this once the proximity monitor refactor is done.

seems like that was a lie

**Pull request closed without merging**